### PR TITLE
Migrate all dev scripts + docs from npm to yarn.

### PR DESCRIPTION
**Pull request by paulirish**
Users are very welcome to install lighthouse with npm, but if they're doing a clone and any local development, all scripts + docs use yarn.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,21 +20,21 @@ cache:
...
-  - npm run install-all
+  - yarn run install-all
 before_script:
   - gem install travis-artifacts
   - export DISPLAY=:99.0
   - export LIGHTHOUSE_CHROMIUM_PATH="$(pwd)/chrome-linux/chrome"
   - sh -e /etc/init.d/xvfb start
   - ./lighthouse-core/scripts/download-chrome.sh
-  - npm run build-all
+  - yarn run build-all
```
Comment (ebidel) [inline .travis.yml]: Pretty sure the "run" is optional. 

`yarn blah` would be great for all the things.

Comment (paulirish) [inline .travis.yml]: ya it is. i was keeping it in some places to be explicit, but really i'd prefer to go without "run", too. :)

i'll nuke them.

Comment (paulirish) [inline .travis.yml]: done

File: readme.md [hunk 2]
```diff
@@ -238,13 +238,12 @@ right corner and signing in to GitHub.
 git clone https://github.com/GoogleChrome/lighthouse
```
Comment (brendankenny) [inline readme.md]: we should add a note about needing `yarn` for development here

Comment (paulirish) [inline readme.md]: sg. done 2bb9c5b31a37bcd7f6c220f2997cd68ba43c86b1

**Pull request merged**